Assets/Experimental/OptionSystem/OptionSystem.cs

```diff
@@ -0,0 +1,230 @@
+        audio = Get<AudioOptions>();
+        audio.Print();
+        some = Get<SomeOptions>();
+        some.Print();
+    }
+
+	void Initialize()
+    {
+        // get all classes that have Options attributes
+        var q = from t in Assembly.GetExecutingAssembly().GetTypes()
+                where t.IsClass && t.GetCustomAttributes(typeof(OptionCategory), true).Length > 0
+                select t;
+
+        StringBuilder allPropertiesStr = new StringBuilder();
+        foreach (var t in q)
+        {
+
+            string categoryClassName = t.FullName;
+            string categoryAttrValue = string.Empty;
+            var attrs = t.GetCustomAttributes(typeof(OptionCategory), true);
+            foreach (var attr in attrs)
+            {
+                if (attr is OptionCategory)
+                {
+                    OptionCategory o = (OptionCategory)attr;
+                    categoryAttrValue = o.Name;
+                }
+            }
+
+            foreach (var p in t.GetProperties())
+            {
+                string propertyName = p.Name;
+                string propertyAttrValue = string.Empty;
+                foreach (var attr in attrs)
+                {
+                    if (attr is Option)
+                    {
+                        Option o = (Option)attr;
+                        propertyAttrValue = o.Name;
+                    }
+                }
+                StringBuilder propertyStr = new StringBuilder();
+                propertyStr.Append(categoryClassName);
+                propertyStr.Append('*');
+                propertyStr.Append(propertyName);
+                if (!PlayerPrefs.HasKey(propertyStr.ToString()))
+                {
+                    InitializePrefProperty(p.PropertyType, propertyStr.ToString());
+                }
+                allPropertiesStr.Append(propertyStr);
+                allPropertiesStr.Append(',');
+            }
+        }
+
+        allPropertiesStr.Remove(allPropertiesStr.Length - 1, 1);
+        if (PlayerPrefs.HasKey(allOptionsKey))
+        {
+            List<string> oldKeys = PlayerPrefs.GetString(allOptionsKey).Split(',').ToList();
+            List<string> currKeys = allPropertiesStr.ToString().Split(',').ToList();
+            var obsoleteKeys = oldKeys.Except(currKeys);
+            foreach (var key in obsoleteKeys)
+            {
+                PlayerPrefs.DeleteKey(key);
+            }
+        }
+
+        PlayerPrefs.SetString(allOptionsKey, allPropertiesStr.ToString());
+
+        PlayerPrefs.Save();
+	}
+
+    T Get<T>()
+    {
+        object obj = optionObjs[typeof(T)];
+        return (T)Convert.ChangeType(obj, typeof(T));
+    }
+
+    void SaveAllChanges()
+    {
+        foreach(var pair in optionObjs)
+        {
+            Type t = pair.Key;
+            string typeName = t.FullName;
+            
+            foreach (var p in t.GetProperties())
+            {
+                StringBuilder keyStr = new StringBuilder();
```

Comment:
For short strings, use string interpolation via ``string.Format`` or just normal concatenation. StringBuilder is more suited for long complex strings.